### PR TITLE
added snapshot mount tc and related functions to tempest

### PR DESCRIPTION
**Pull request by maheshbhoge**
Added snapshot mount testcases to tempest.  wrote some functions/libraries to automate this scenarios.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Repo for automation build, test etc.
         - volume_size = 1 → Size of the volume
         - volume_type_1 = lvm → LVM Volume type name for volume creation
         - volume_type_id_1 = d6cceecf-a5b8-4b32-995f-3a1e1271ca28 → ID of LVM volume type specified in volume_type_1 field
+        - fvm_image_ref = a7cceecf-a5b8-4b32-995f-3a1e1271df23 -> ID of file manager instance(FVM), reqiured to run snapshot mount tc.
```
Comment (savitha-p) [inline README.md]: Please add this field under compute section.

File: tempest/api/workloadmgr/base.py [hunk 2]
```diff
@@ -2420,3 +2420,45 @@ def restart_wlm_api_service(self):
         ssh.close()
         return status_update
 
+    '''
+    Method returns the Instance ID of a new VM instance created
+    '''
+    def create_file_manager_vm(self, vm_cleanup=True, vm_name="", security_group_id = "default", flavor_id =CONF.compute.flavor_ref, \
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py]: Why is a different method required for this ? Can the existing one be reused?

File: tempest/tvaultconf.py [hunk 3]
```diff
@@ -43,15 +43,15 @@
 restore_type = "restore"
 global_job_scheduler=False
 
-tvault_ip = "192.168.1.23"
+tvault_ip = "192.168.12.145"
 tvault_dbusername = "root"
 tvault_dbname = "workloadmgr"
-tvault_dbpassword = "sample-password"
+tvault_dbpassword = "52T8FVYZJse"
```
Comment (savitha-p) [inline tempest/tvaultconf.py]: Please revert this field, as its intended to keep sample password here.

File: tempest/tvaultconf.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 #If you want to cleanup all test resources like vms, volumes, workloads then set
 # following cleanup parameter value to True otherwise False
-cleanup = True
+cleanup = False
```
Comment (savitha-p) [inline tempest/tvaultconf.py]: Always checkin cleanup=True by default. You can change it False locally for your testing.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+LOG = logging.getLogger(__name__)
+CONF = config.CONF
+
+class WorkloadsTest(base.BaseWorkloadmgrTest):
+
+    credentials = ['primary']
+
+    instances_ids = []
+    snapshot_ids = []
+    date_from = ""
```
Comment (savitha-p) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Where are the parameters "date_from", "date_to" used ?

File: README.md (hunk 1 above)
Comment (savitha-p) [reply README.md]: Is this done?

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py (hunk 5 above)
Comment (savitha-p) [reply tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Remove any unused variables.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+	    global instances_ids
+	    global snapshot_ids 
+	    global wid
+	    global security_group_id
+	    global volumes_ids
+            global fvm_id
+	    instances_ids = self.instances_ids
+            snapshot_ids = self.snapshot_ids
+	    date_from = self.date_from
```
Comment (savitha-p) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Remove any unused variables.

File: tempest/prerequisites.py [hunk 7]
```diff
@@ -661,3 +661,105 @@ def nested_security(self):
...
+        self.wait_for_workload_tobe_available(self.wid)
+        if(self.getSnapshotStatus(self.wid, self.snapshot_id) != "available"):
+            self.exception = "Create full snapshot"
+            raise Exception(str(self.exception)) 
+
+	self.snapshot_ids.append(self.snapshot_id)
+
+        LOG.debug("Snapshot ID-1: " + str(self.snapshot_ids[0]))
+        time_now = time.time()
+        self.date_from = datetime.datetime.utcfromtimestamp(time_now).strftime("%Y-%m-%dT%H:%M:%S")
```
Comment (savitha-p) [inline tempest/prerequisites.py]: Is this required?

File: tempest/prerequisites.py [hunk 8]
```diff
@@ -661,3 +661,105 @@ def nested_security(self):
...
+        self.wait_for_workload_tobe_available(self.wid)
+        if(self.getSnapshotStatus(self.wid, self.snapshot_id) != "available"):
+            self.exception = "Create incremental-1 snapshot"
+            raise Exception(str(self.exception))
+
+	self.snapshot_ids.append(self.snapshot_id)
+        LOG.debug("Snapshot ID-2: " + str(self.snapshot_ids[1]))    
+
+        time_now = time.time()
+        self.date_to = datetime.datetime.utcfromtimestamp(time_now).strftime("%Y-%m-%dT%H:%M:%S")
```
Comment (savitha-p) [inline tempest/prerequisites.py]: Do we need to capture this timestamp?

File: README.md (hunk 1 above)
Comment (maheshbhoge) [reply README.md]: Done

File: tempest/api/workloadmgr/base.py (hunk 2 above)
Comment (maheshbhoge) [reply tempest/api/workloadmgr/base.py]: Removed method

File: tempest/tvaultconf.py (hunk 3 above)
Comment (maheshbhoge) [reply tempest/tvaultconf.py]: reverted tvaultconf.y file

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py (hunk 5 above)
Comment (maheshbhoge) [reply tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: removed unused varialbes

File: tempest/prerequisites.py (hunk 7 above)
Comment (maheshbhoge) [reply tempest/prerequisites.py]: removed timestamp functions

**Comment by savitha-p** [comment]
Please add more scenarios as mentioned by Murali on confluence in next PR.

**Pull request merged**